loader/src/ui/nodes/TextArea.cpp

```diff
@@ -0,0 +1,229 @@
+#include <Geode/ui/TextArea.hpp>
+
+using namespace geode::prelude;
+
+SimpleTextArea* SimpleTextArea::create(const std::string& font, const std::string& text, const float scale = 1) {
+    return SimpleTextArea::create(font, text, scale, 500, false);
+}
+
+SimpleTextArea* SimpleTextArea::create(const std::string& font, const std::string& text, const float scale, const float width) {
+    return SimpleTextArea::create(font, text, scale, width, true);
+}
+
+SimpleTextArea* SimpleTextArea::create(const std::string& font, const std::string& text, const float scale, const float width, const bool artificialWidth) {
+    SimpleTextArea* instance = new SimpleTextArea(font, text, scale, width, artificialWidth);
+
+    if (instance && instance->init()) {
+        instance->autorelease();
+
+        return instance;
+    } else {
+        CC_SAFE_DELETE(instance);
+        
+        return nullptr;
+    }
+}
+
+SimpleTextArea::SimpleTextArea(const std::string& font, const std::string& text, const float scale, const float width, const bool artificialWidth) {
+    this->m_font = font;
+    this->m_text = text;
+    this->m_maxLines = 0;
+    this->m_scale = scale;
+    this->m_linePadding = 0;
+    this->m_artificialWidth = artificialWidth;
+    this->m_container = CCMenu::create();
+
+    this->setAnchorPoint({ 0.5f, 0.5f });
+    this->m_container->setPosition({ 0, 0 });
+    this->m_container->setAnchorPoint({ 0, 1 });
+    this->m_container->setContentSize({ width, 0 });
+
+    this->addChild(this->m_container);
+    this->updateContents();
+}
+
+void SimpleTextArea::setFont(const std::string& font) {
+    this->m_font = font;
+
+    this->updateContents();
+}
+
+std::string SimpleTextArea::getFont() {
+    return this->m_font;
+}
+
+void SimpleTextArea::setAlignment(const CCTextAlignment alignment) {
+    this->m_alignment = alignment;
+
+    this->updateContents();
+}
+
+CCTextAlignment SimpleTextArea::getAlignment() {
+    return this->m_alignment;
+}
+
+void SimpleTextArea::setText(const std::string& text) {
+    this->m_text = text;
+
+    this->updateContents();
+}
+
+std::string SimpleTextArea::getText() {
+    return this->m_text;
+}
+
+void SimpleTextArea::setMaxLines(const size_t maxLines) {
+    this->m_maxLines = maxLines;
+
+    this->updateContents();
+}
+
+size_t SimpleTextArea::getMaxLines() {
+    return this->m_maxLines;
+}
+
+void SimpleTextArea::setWidth(const float width) {
+        this->m_artificialWidth = true;
+
+        this->setContentSize({ width, this->getContentSize().height });
+        this->m_container->setContentSize({ width, this->m_container->getContentSize().height });
+    }
```

Comment:
oops

Reply:
both the container and node itself need to maintain the same content size to prevent weird overflowing behavior

Reply:
no i just mean the function has the wrong indentation